### ISSUE TITLE
Non-Convex Faces Not Triangulated Correctly

### ISSUE THREAD

**Issue by maxattack** [nem0/OpenFBX#56]
The triangulate() helper function in ofbx.cpp doesn't properly ear-cut non-convex faces, leading to non-internal edges being placed.

![image](https://user-images.githubusercontent.com/798685/85236948-6611bb00-b3d7-11ea-896e-d18e42f1f47c.png)
![image](https://user-images.githubusercontent.com/798685/85236949-69a54200-b3d7-11ea-8dd9-cc3f3b5334ff.png)

**Comment by maxattack** [comment]
The algorithm is not too tricky -- it's called "ear-cutting." A reference: http://www-cgrl.cs.mcgill.ca/~godfried/teaching/cg-projects/97/Ian/algorithm1.html

I was able to implement this is ~200LOC

**Comment by nem0** [comment]
Hi, I'm not sure when will I get to this, but if you need this soon you can disable ofbx triangulation and and do correct triangulation on your side.

**Comment by maxattack** [comment]
Yeah, that's what I've already implemented. Just putting it on your radar :)

**Comment by maxattack** [comment]
For the curious: https://github.com/maxattack/Trinket/blob/master/Source/Triangulate.cpp
